### PR TITLE
[Distributed] [15/N] Fix clang-tidy warnings in torch/csrc/distributed/

### PR DESCRIPTION
**Pull request by cyyever**
Follows  #136848

cc @XilunWu @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

### CODE REVIEW

File: torch/csrc/distributed/rpc/script_call.cpp [hunk 1]
```diff
@@ -9,11 +9,13 @@ const std::string ScriptCall::ATEN_PREFIX_("aten::");
 
 ScriptCall::ScriptCall(
     std::shared_ptr<Operator> op,
+    // NOLINTNEXTLINE(cppcoreguidelines-rvalue-reference-param-not-moved)
     std::vector<at::IValue>&& stack)
     : op_(std::move(op)), stack_(stack), isAsyncExecution_(false) {}
```
Comment (Skylion007) [inline torch/csrc/distributed/rpc/script_call.cpp:14]: Why not move these parameters here? Seems like pretty clearly should be a move into `stack_`?

File: torch/csrc/distributed/autograd/functions/recvrpc_backward.cpp [hunk 2]
```diff
@@ -4,23 +4,22 @@
...
-namespace autograd {
+namespace torch::distributed::autograd {
 
 using torch::autograd::Variable;
 using torch::autograd::variable_list;
 
 RecvRpcBackward::RecvRpcBackward(
     const AutogradMetadata& autogradMetadata,
-    ContextPtr autogradContext,
+    const ContextPtr& autogradContext,
```
[the remark this answers is not in the document]
Comment (cyyever) [reply torch/csrc/distributed/autograd/functions/recvrpc_backward.cpp:14]: After checked the type, I found it is std::shared_ptr passed to std::weak_ptr.

File: torch/csrc/distributed/rpc/script_call.cpp (hunk 1 above)
Comment (cyyever) [reply torch/csrc/distributed/rpc/script_call.cpp:14]: A previous commit using std::move here caused test core dump, it needs further investigation.

Comment (Skylion007) [reply torch/csrc/distributed/rpc/script_call.cpp:14]: @cyyever Sounds like that is probably caused by a use after free wherever the ctor is called?

Comment (cyyever) [reply torch/csrc/distributed/rpc/script_call.cpp:14]: I think so.

**Review by kwen2501 — APPROVED** [review]
Thanks

**Pull request closed without merging**